Automattic/jetpack
# Update podcast cache timeout to 1 hour

#### Changes proposed in this Pull Request:

* This PR expands on the work in https://github.com/Automattic/jetpack/pull/24966 and defaults the RSS feed cache timeout to 1 hour for podcasts used by the Anchor.fm or Podcast Player blocks.

#### Other information:


#### Jetpack product discussion

p1HpG7-h2Q-p2

#### Does this pull request change what data or activity we track or use?

No changes to tracking or data collection.

#### Testing instructions:

* Ensure that you have the Jetpack plugin installed, connected, and activated
* Ensure that you have the Jetpack Beta plugin installed and activated
* Ensure that you have the Gutenberg plugin installed and activated
* Navigate to Jetpack -> Jetpack Beta in the left nav
* Enter `update/podcast-cache-timeout-to-1-hour` into the "Search for a Jetpack Feature Branch" search field, and click on the Activate button next to the branch when it appears
* Add the following code to your site, either manually or using a tool like [Code Snippets](https://wordpress.org/plugins/code-snippets/):
```php
add_filter( 'jetpack_podcast_feed_cache_timeout', function ( $cache_timeout, $feed_url ) {
	error_log( "jetpack_podcast_feed_cache_timeout filter; \$feed_url=$feed_url; \$cache_timeout=$cache_timeout" );
	return $cache_timeout;
} );

add_action( 'setted_transient', function ( $transient_name, $transient_value, $expiration ) {
	// We want to log sets to `feed_*` and `feed_mod_*`
	if ( 0 !== strpos( $transient_name, 'feed_' ) ) {
		return;
	}

	error_log( "Setting RSS feed transient $transient_name with expiration $expiration" );
}, 10, 3 );
```
* Ensure you have a way to view your PHP error log, either via file system access, or via a tool like [Error Log Monitor](https://wordpress.org/plugins/error-log-monitor/).
* Navigate to _Posts_ -> _Add New_
* In the editor, add a new podcast player block
* In the input field, enter the URL of a podcast RSS feed that you haven't used previously. You can use a URL like `https://distributed.blog/category/podcast/feed/?somerandomsuffix873287` to ensure the feed hasn't been cached.
* View your PHP error log, and verify that you see three distinct entries like:
```
jetpack_podcast_feed_cache_timeout; $feed_url=https://distributed.blog/category/podcast/feed/?abcyxzedf; $cache_timeout=3600
Setting RSS feed transient feed_[hash] with expiration 3600
Setting RSS feed transient feed_mod_[hash] with expiration 3600
```
